deploy-0chain/action.yml

```diff
@@ -176,61 +170,11 @@ runs:
         helm upgrade --install 0proxy --set zproxy.hostName=${{ env.NAMESPACE }} --set zproxy.host=devnet-0chain.net --set zproxy.config.blockWorker=https://${{ env.NAMESPACE }}.devnet-0chain.net/dns --set zproxy.image.tag=${{ env.ZPROXY_TAG }} -n ${{ env.NAMESPACE }}  0chain-helm/zproxy --kubeconfig ./kube/${{ env.NAMESPACE }}-config
         helm upgrade --install 0box --wait --set zbox.hostName=${{ env.NAMESPACE }} --set zbox.host=devnet-0chain.net --set zbox.config.blockWorker=https://${{ env.NAMESPACE }}.devnet-0chain.net/dns --set zbox.image.tag=${{ env.ZBOX_TAG }} -n ${{ env.NAMESPACE }}  0chain-helm/zbox --kubeconfig ./kube/${{ env.NAMESPACE }}-config
         helm upgrade --install 0search --set zsearch.hostName=${{ env.NAMESPACE }} --set zsearch.host=devnet-0chain.net --set zsearch.image.tag=${{ env.ZSEARCH_TAG }} -n ${{ env.NAMESPACE }}  0chain-helm/zsearch --kubeconfig ./kube/${{ env.NAMESPACE }}-config
-        helm upgrade --install blobber-stake --wait -n ${{ env.NAMESPACE }} --set blobberStake.config.blockWorker=https://${{ env.NAMESPACE }}.devnet-0chain.net/dns --set blobberStake.blobberCount=6 --set blobberStake.hostName=${{ env.NAMESPACE }} --set blobberStake.host=devnet-0chain.net --set blobberStake.image.tag=${{ env.BLOBBER_STAKE_TAG }} 0chain-helm/blobberStake --kubeconfig ./kube/${{ env.NAMESPACE }}-config
         helm upgrade --install authorizer -n ${{ env.NAMESPACE }} --set authorizer.authorizerCount=3 --set authorizer.hostName=${{ env.NAMESPACE }} --set authorizer.host=devnet-0chain.net --set authorizer.image.tag=${{ env.AUTHORIZER_TAG }} --set authorizer.config.blockWorker=https://${{ env.NAMESPACE }}.devnet-0chain.net/dns 0chain-helm/authorizer --kubeconfig ./kube/${{ env.NAMESPACE }}-config
         ' > ./deploy.sh;
         chmod 777 ./deploy.sh;
         ./deploy.sh
 
-    - name: "Wait for 0Chain network deployment to complete"
```

Comment:
I would still have this - sometimes the miners do not register and we need to re-attempt the deployment

Reply:
This essentially only waited for `Blobber-stake` pod to reach `completed` right? The other pods all have `--wait` in them so I guess they are completed by the time `./deploy.sh` is exited. In the runs I did deployment is always done by the time ./deploy.sh finishes.
Re: miners not registering I think it has in-built retries, I have seen them go from `Error` back to `ContainerCreating`. 

All being said I'm aware we don't want false negative on test runs and that many things can cause failure in deployment, so if you could give me an idea of what sort of wait/retries we want I'll be happy to add it

Reply:
I would keep what was here only instead of waiting for stake container to finish we try your new approach instead.
if your approach fails we retry internally say 5 times 
after that, we do a full network redeployment

Reply:
Done. Here's what a purposefully failed run looks like now: https://github.com/0chain/system_test/runs/5727416719?check_suite_focus=true